CMakeLists.txt

```diff
@@ -606,10 +605,18 @@ if(GPERF)
   find_package(Gperftools REQUIRED)
 endif()
 
-if(USE_GETTEXT)
-  find_package(Gettext)
+find_package(Gettext)
+if (Gettext_FOUND)
+  set(PO_COUNT 0)
   foreach(lang bg da de es fr hr it ko_KR pt_BR ru sv zh_CN zh_TW)
     GETTEXT_PROCESS_PO_FILES(${lang} ALL PO_FILES ${CMAKE_CURRENT_LIST_DIR}/Translations/${lang}.po)
+
+    if(ANDROID)
+      math(EXPR PO_COUNT "${PO_COUNT}+1")
+      add_custom_target(copy_${lang} ALL COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_CURRENT_BINARY_DIR}/${lang}.gmo" "${DevilutionX_SOURCE_DIR}/android-project/app/src/main/assets/${lang}.gmo")
+      add_dependencies(${BIN_TARGET} copy_${lang})
+      add_dependencies(copy_${lang} "pofiles_${PO_COUNT}")

```

Comment:
What's this for? As far as I can tell this `pofiles_${PO_COUNT}` doesn't seem to do anything useful

Reply:
Android studio /grade doesn't build the translation targets, so this is adding them as dependencies so that they get build with the project

Reply:
It doesn't though, `"pofiles_${PO_COUNT}"` is not a target that exists. I don't think you'll need after fixing the commands per the comment below. What I think happens here is a rebuild triggers every time because it depends on this non-existing target.

Reply:
GETTEXT_PROCESS_PO_FILES generates the pofiles_# targets, I needed the dependency to even generate the gmo files even before I started looking in to relocating them to the assets folder.
But I'll give removing it a shot just in case 🤷

Reply:
GETTEXT_PROCESS_PO_FILES also generates/updates `pofiles` target with dep set to all `pofiles_N`
wouldn't it be easier/better to just copy all .gmo files at once (via custom command, that depends on `pofiles` target)

Reply:
I like @isage's idea. How about this?

```cmake
find_package(Gettext)
if (Gettext_FOUND)
  foreach(lang bg da de es fr hr it ko_KR pt_BR ru sv zh_CN zh_TW)
    GETTEXT_PROCESS_PO_FILES(${lang} ALL PO_FILES ${CMAKE_CURRENT_LIST_DIR}/Translations/${lang}.po)
    list(APPEND libdevilutionx_TRANSLATIONS ${CMAKE_CURRENT_BINARY_DIR}/${lang}.gmo)
  endforeach(lang)
  
  if(ANDROID)
    add_custom_target(copy_translations ALL
      COMMAND ${CMAKE_COMMAND} -E copy ${libdevilutionx_TRANSLATIONS} "${DevilutionX_SOURCE_DIR}/android-project/app/src/main/assets"
      DEPENDS ${libdevilutionx_TRANSLATIONS})
      
    add_dependencies(${BIN_TARGET} copy_translations)
  endif()
endif()
```

Reply:
> GETTEXT_PROCESS_PO_FILES also generates/updates `pofiles` target with dep set to all `pofiles_N`
> wouldn't it be easier/better to just copy all .gmo files at once (via custom command, that depends on `pofiles` target)

I would think so too from looking at FindGettext.cmake, but this is what I get if I set the target to `pofiles`:
`> Unexpected native build target pofiles. Valid values are: libdevilutionx, StormLib, pofiles_8, fmt, pofiles_7, pofiles_9, pofiles_4, pofiles_3, SDL2, pofiles_6, pofiles_5, uninstall, devilutionx, PKWare, pofiles_2, pofiles_1, SDL2_ttf, hidapi, pofiles_10, pofiles_11, SDL_audiolib, pofiles_12, pofiles_13, smacker, sodium, freetype, SDL2main`

Reply:
> How about this?

Works nicely, lets go with that. Boy do I feel useless now :unamused: